packages/interactivity-router/src/index.js

```diff
@@ -246,3 +317,34 @@ export const { state, actions } = store( 'core/router', {
 		},
 	},
 } );
+
+// Add click and prefetch to all links.
+if ( process.env.IS_GUTENBERG_PLUGIN ) {
+	if ( navigationMode === 'fullPage' ) {
+		// Navigate on click.
+		document.addEventListener(
+			'click',
+			function ( event ) {
+				const ref = event.target.closest( 'a' );
+				if ( isValidLink( ref ) && isValidEvent( event ) ) {
+					event.preventDefault();
+					actions.navigate( ref.href );
+				}
+			},
+			true
+		);
+		// Prefetch on hover.
+		document.addEventListener(
+			'mouseenter',
```

Comment:
On touch interfaces, I understand that the `mouseenter` event does fire but after a 300 ms delay. To ensure that prefetch happens immediately on a touch device, I suggest adding the same handler for the `touchstart` event. Or apparently `pointerenter` will handle both of these and it is supported by all browsers: https://caniuse.com/mdn-api_element_pointerenter_event